creds/ssm/manager.go

```diff
@@ -0,0 +1,123 @@
+package ssm
+
+import (
+	"errors"
+	"io/ioutil"
+	"os"
+	"text/template"
+	"text/template/parse"
+
+	"code.cloudfoundry.org/lager"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/concourse/atc/creds"
+)
+
+const DefaultPipeSecretTemplate = "/concourse/{{.Team}}/{{.Pipeline}}/{{.Secret}}"
+const DefaultTeamSecretTemplate = "/concourse/{{.Team}}/{{.Secret}}"
+
+type SsmManager struct {
+	AwsAccessKeyID     string `long:"aws-access-key" description:"AWS Access key ID"`
+	AwsSecretAccessKey string `long:"aws-secret-key" description:"AWS Secret Access Key"`
+	AwsSessionToken    string `long:"aws-session-token" description:"AWS Session Token"`
+	AwsRegion          string `long:"aws-region" description:"AWS region to send requests to. Enviroment variable AWS_REGION is used if this flag is not provided."`
```

Comment:
This can be done as `env:"AWS_REGION"` in the struct tags. (You can then remove that blurb from the description; I think it'll show up in the help-text automatically.)